presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -363,6 +373,40 @@ void PrestoServer::run() {
         json infoStateJson = convertNodeState(server->nodeState());
         http::sendOkResponse(downstream, infoStateJson);
       });
+  httpServer_->registerPut(
+    "/v1/info/state",
+    [server = this](
+        proxygen::HTTPMessage* /*message*/,
+        const std::vector<std::unique_ptr<folly::IOBuf>>& body ,
+        proxygen::ResponseHandler* downstream) {
+         std::string requestBody=getRequestBody(body);
+         try {
+            folly::dynamic jsonBody = folly::parseJson(requestBody);
+            if (jsonBody == "SHUTTING_DOWN") {
+                // Print message and initiate shutdown
+                LOG(INFO) << "Shutdown requested";
+                if (server->nodeState() == NodeState::kActive) {
+                    // Call stop() to handle the shutdown process
+                    std::thread([server]() {
+                            server->stop();
+                        }).detach();
+                } else {
+                    // Print message and indicate that shutdown is already requested
```

Comment:
State could be `kInActive` as well. So we need to say `Node is inactive or shutdown already requested`

Reply:
Done.

Reply:
Did you push this change?

Reply:
Actually, I am having issues in using try_parse(), it expects an array as an input something like:  [
            { "op": "replace", "path": "/name", "value": "Jane Doe" },
            { "op": "add", "path": "/location", "value": "New York" }
        ] but our requestBody just contains a string SHUTTING_DOWN so when I am passing it in the function it is throwing parse error. You can see in the attached image I am parsing a string named patchStr and requestBody, patchStr is parsed without am issue whereas requestBody throws parse error (i.e. jsonPatchResult2.hasValue() is 0).
<img width="1728" alt="Screenshot 2024-10-04 at 7 19 47 PM" src="https://github.com/user-attachments/assets/651e3340-2b71-43ea-b9c5-7f728875e633">
May I move back to using try-catch ?

Reply:
I am looking at the folly API on how to use this. There is a test here https://github.com/facebook/folly/blob/e08c228878aaff37aa6abc348154bd5a3e1e82fb/folly/json/test/json_patch_test.cpp#L86

Reply:
> but our requestBody just contains a string SHUTTING_DOWN so when I am passing it in the function it is throwing parse error

The request is not a json string right? Why parse as json then? Just compare the body directly?